Support/Scripts/prepare-android-ndk.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+##
+## Copyright (c) 2014-present, Facebook, Inc.
+## All rights reserved.
+##
+## This source code is licensed under the University of Illinois/NCSA Open
+## Source License found in the LICENSE file in the root directory of this
+## source tree. An additional grant of patent rights can be found in the
+## PATENTS file in the same directory.
+##
+
+import sys
+import argparse
+from typing import Any
+import urllib.request
+import zipfile
+import os
+import shutil
+
+try:
+    import progress.bar
+except ImportError:
+    print("python3 -m pip install progress to see progress bars")
+
+bar = None
+bar_message = None
+
+
+def update_progress_bar(chunk_number: int, maximum_chunks: int, total_size: int):
+    global bar
+    if "progress.bar" not in sys.modules:
+        return
+    if bar is None:
+        bar = progress.bar.Bar(bar_message, max=total_size / maximum_chunks)
+
+    bar.next()
+
+
+def unzip(source: str, destination: str):
+    def unzip_file(zf, info, destination):
+        dest_path = zf.extract(info.filename, path=destination)
+
+        perm = info.external_attr >> 16
+        os.chmod(dest_path, perm)
+
+    with zipfile.ZipFile(source, "r") as f:
+        global bar
+        bar = progress.bar.Bar(bar_message, max=len(f.infolist()))
+        for info in f.infolist():
+            bar.next()
+            unzip_file(f, info, destination)
+
+
+def parse_arguments() -> argparse.Namespace:
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--download-prefix",
+        help="the directory prefix for the ndk installation",
+        default="/tmp",
+    )
+
+    args = parser.parse_args()
+    return args
+
+
+def main():
+    global bar_message
+    args = parse_arguments()
+
+    if sys.platform == "linux":
+        host = "linux-x86_64"
+    elif sys.platform == "darwin":
+        host = "darwin-x86_64"
+    else:
+        print("Uninmplemented platform: {}".format(sys.platform))
+        sys.exit(1)
+
+    ndk_version = "r16b"
+    ndk_package_base = "android-ndk-{}".format(ndk_version)
+    ndk_package_archive = "{}-{}.zip".format(ndk_package_base, host)
+    ndk_url = "https://dl.google.com/android/repository/{}".format(ndk_package_archive)
+    ndk_download_path = "{}/{}".format(args.download_prefix, ndk_package_archive)
+    ndk_tmp_unzip = "/tmp/android-ndk-unzip"
+    ndk_unzip_path = "{}/android-ndk".format(args.download_prefix)
+
+    bar_message = "Downloading NDK"
+    urllib.request.urlretrieve(ndk_url, ndk_download_path, update_progress_bar)
+    bar.finish()
+
+    bar_message = "Unzipping NDK"
```

Comment:
I don't think this will get updated because there isn't any call to `update_progress_bar()` after `urlretrieve()` completes.